airbnb_project/listings/harvester_app/harvester/pipelines.py

```diff
@@ -102,3 +105,80 @@ def _extract_bathrooms(self, adapter):
+                - airbnb_listing_id (str): Unique identifier for the Airbnb listing
+                - name (str): Name of the listing
+                - title (str): Title of the listing
+                - baths (str): Number of bathrooms
+                - beds (str): Number of beds
+                - latitude (str): Latitude coordinate
+                - longitude (str): Longitude coordinate
+                - person_capacity (str): Maximum number of guests
+                - registration_number (str): Official registration number
+                - room_type (str): Type of room/accommodation
+                - location (str): Location description
+                - bath_is_shared (bool): Whether bathroom is shared
+                - baths_text (str): Textual description of bathroom facilities
+            spider: The spider instance that is running the crawl
+
+        Returns:
+            dict: The original item dictionary, unmodified
+
+        Note:
+            - All fields are required - missing or empty airbnb_listing_id will cause the item to be logged and skipped
+            - If a listing with the same airbnb_listing_id exists, it will be skipped
+        """
+        airbnb_listing_id = item.get('airbnb_listing_id', '')
+
+        # Early return if no valid airbnb_listing_id
+        if not airbnb_listing_id:
+            spider.logger.error("Missing required airbnb_listing_id, skipping item")
+            return item
+        try:
+            # Check if the listing already exists
+            if not Listing.objects.filter(airbnb_listing_id=item.get('airbnb_listing_id')).exists():
+                # Create and save the new listing if it doesn't exist
+                listing = Listing(
+                    airbnb_listing_id=item.get('airbnb_listing_id'),
+                    name=item.get('name'),
+                    title=item.get('title'),
+                    baths=item.get('baths'),
+                    beds=item.get('beds'),
+                    latitude=item.get('latitude'),
+                    longitude=item.get('longitude'),
+                    person_capacity=item.get('person_capacity'),
+                    registration_number=item.get('registration_number'),
+                    room_type=item.get('room_type'),
+                    location=item.get('location'),
+                    is_bath_shared=item.get('bath_is_shared'),
+                    baths_text=item.get('baths_text')
+                )
+                # Save the listing to the database
+                listing.save()
+                spider.logger.info(f"New listing {item.get('airbnb_listing_id')} saved to the database.")
+            else:
+                # Log if the listing already exists and is skipped
+                spider.logger.info(f"Listing {item.get('airbnb_listing_id')} already exists, skipping.")
+        except IntegrityError as e:
+            spider.logger.error(f"Failed to save listing {item.get('name')} to the database: {e}")
```

Comment:
can the name be undefined/None ?

Reply:
I think it can be empty maybe not undefined/None, It will be better to log the `airbnb_listing_id`.

Reply:
thank you!